src/charm.py

```diff
@@ -453,6 +449,36 @@ def _on_database_changed(self, event: pgsql.DatabaseChangedEvent) -> None:  # ty
 
         self._config_changed(event)
 
+    def _on_add_admin_user_action(self, event: ActionEvent) -> None:
+        """Add a new admin user to Discourse.
+
+        Args:
+            event: Event triggering the add_admin_user action.
+        """
+        email = event.params["email"]
+        password = event.params["password"]
+        container = self.unit.get_container("discourse")
+        if container.can_connect():
+            process = container.exec(
+                [
+                    "bash",
+                    "-c",
+                    f"./bin/bundle exec rake admin:create <<< $'{email}\n{password}\n{password}\nY'",
+                ],
+                user="discourse",
+                working_dir=DISCOURSE_PATH,
+                environment=self._create_discourse_environment_settings(),
+            )
+            try:
+                process.wait_output()
+            except ExecError as ex:
+                logger.error(ex)
+                event.fail(
+                    # Parameter validation errors are printed to stdout
+                    f"Failed to create user with email {email}: {ex.stdout}"  # type: ignore
+                )
+            event.set_results({"user": f"{email}"})
```

Comment:
Should we set this result if the action fails ?

Reply:
Fixed. Thanks